src/backend.ts

```diff
@@ -0,0 +1,101 @@
+import { BrowserOptions } from '@sentry/browser';
+import { BrowserBackend } from '@sentry/browser/dist/backend';
+import { BaseBackend, getCurrentHub } from '@sentry/core';
+import { forget, getGlobalObject } from '@sentry/utils';
+
+import { Capacitor, Plugins } from '@capacitor/core';
+
+import { CapacitorOptions } from './options';
+
+const SentryCapacitor = Plugins;
+
+/**
+ * The Sentry Capacitor SDK Backend.
+ */
+export class CapacitorBackend extends BaseBackend<BrowserOptions> {
+  private readonly _browserBackend: BrowserBackend;
+
+  /**
+   * Creates a new Capacitor backend instance.
+   */
+  public constructor(protected readonly _options: CapacitorOptions = {}) {
+    super(_options);
+    this._browserBackend = new BrowserBackend(_options);
+
+    if (this._isCapacitor() && _options.enableNative !== false) {
+      this._runNativeInstall();
+    }
+  }
+
+  // CAPACITOR -----------------
+  /**
+   * Uses reference to the plugin to call Capacitor functions
+   * @param action name of the action
+   * @param args Arguments
+   */
+  private async _nativeCall(action: string, ...args: any[]): Promise<void> {
+    return new Promise<any>((resolve, reject) => {
+      if (this._options.enableNative === false) {
+        reject('enableNative = false, using browser transport');
+        return;
+      }
+
+      try {
+        window['SentryCapacitor'][`${action}`](args);
+      } catch (error) {
+        reject('Capacitor is not available. ' + error);
+      }
+    });
+  }
+
+  /**
+   * Calling into native install function
+   */
+  private _runNativeInstall(): void {
+    if (this._options.dsn && this._options.enabled !== false) {
+      forget(this._nativeCall('install', this._options.dsn, this._options));
+    }
+
+    const scope = getCurrentHub().getScope();
+    if (scope) {
+      scope.addScopeListener(internalScope => {
+        this._nativeCall(
+          'setExtraContext',
+          (internalScope as any)._extra,
+        ).catch(() => {
+          // We do nothing since scope is handled and attached to the event.
+          // This only applies to android.
+        });
+        this._nativeCall('setTagsContext', (internalScope as any)._tags).catch(
+          () => {
+            // We do nothing since scope is handled and attached to the event.
+            // This only applies to android.
+          },
+        );
+        this._nativeCall('setUserContext', (internalScope as any)._user).catch(
+          () => {
+            // We do nothing since scope is handled and attached to the event.
+            // This only applies to android.
+          },
+        );
+        this._nativeCall(
+          'addBreadcrumb',
+          (internalScope as any)._breadcrumbs.pop(),
+        ).catch(() => {
+          // We do nothing since scope is handled and attached to the event.
+          // This only applies to android.
```

Comment:
I think we can delete this line since this should apply to both iOS and Android (since iOS will have envelopes soon)?